### PR TITLE
refactor: add editable featuers to table

### PR DESCRIPTION
**Pull request by yusualhashash**

### CODE REVIEW

File: src/molecules/tables/tables.stories.tsx [hunk 1]
```diff
@@ -278,3 +278,17 @@ export const MultipleActions: StoryObj<typeof Table> = {
     layout: 'centered',
   },
 };
+export const Editable: StoryObj<typeof Table> = {
+  args: {
+    editable: false,
```
Comment (a0m0rajab) [inline src/molecules/tables/tables.stories.tsx:283]: If the story named editable, should not this flag be true?

**Comment by a0m0rajab** [comment]
The delete action does not work on the table ui we should enable it.

File: src/molecules/tables/index.tsx [hunk 2]
```diff
@@ -124,54 +137,19 @@ const ActionComponent = ({
...
- *     callback: () => {},
- *     excludeList: ['id'],
- *     onDelete: () => {},
- *     onEdit: () => {},
- *     tableType: 'UserTable',
- *   },
- * };
- *
- * <DataTable
- *   columnsData={columnsConfig}
```
[the remark this answers is not in the document]
Comment (yusualhashash) [reply src/molecules/tables/index.tsx:160]: done

File: src/molecules/tables/index.tsx [hunk 3]
```diff
@@ -124,54 +137,19 @@ const ActionComponent = ({
...
- * />
- */
+
 export default function DataTable<TData, TValue>({
   columnsData,
   data,
   filterBy,
   action,
-  isLoading,
+  isLoading = false,
```
[the remark this answers is not in the document]
Comment (yusualhashash) [reply src/molecules/tables/index.tsx:146]: done

File: src/molecules/tables/index.tsx [hunk 4]
```diff
@@ -186,19 +164,28 @@ export default function DataTable<TData, TValue>({
...
   const [rowSelection, setRowSelection] = useState({});
 
+  useEffect(() => {
+    if (isLoading) {
+      setTableData(Array(6).fill({}) as TData[]);
+    } else {
+      setTableData(data);
+    }
+  }, [isLoading, data]);
+
```
[the remark this answers is not in the document]
Comment (yusualhashash) [reply src/molecules/tables/index.tsx:174]: because i taked re render error so i used useaffeact

**Pull request merged**